culture/review_culture.md

```diff
@@ -0,0 +1,142 @@
+project. They are the stewards of the review process, and have the privileges
+needed to move forward or block a change. Not everyone can be a maintainer; it
+requires deep knowledge of the project, high trust from the community,
+experience as both a contributor and reviewer, and substantial commitment.
+Since we invest maintainers with special privileges and trust, we have high
+expectations of them. Those acting as maintainers must be respectful,
+considerate, and judicious in the way they exercise their powers.
+
+## Lifecycle of a Review
+
+### Proposal
+The first step is for a contributor to propose a _change_.
+The contributor needs to do two things at this stage:
+- Flesh out the change (e.g. writing the code), so that there is something to review
+- Explain the change (e.g. writing a pull request description), so that reviewers can easily understand it
+
+Both steps are vital, and contributors should put real effort into explaining the change well.
+Doing so is respectful of reviewers' time, and makes it easier for others in the community to follow
+along with the project and get involved themselves.
+
+Finally, before requesting a review, the contributor should _self-review_ their change.
+It's amazing how many issues can be spotted and fixed during self-review.
+This is respectful of reviewers' time, and makes the process a lot smoother.
+
+### Review
+
+Once a change is ready and has been self-reviewed, it's ready for review. Since
+most SourceCred artifacts are tracked via repositories on GitHub, this will
+usually mean creating a pull request.
+
+Once the change is proposed, anyone in the project can add their own review.
+The person proposing the change can also request specific reviewers, if certain
+individiuals have context or knowledge thet will make their reivew particularly
+valuable.
+
+On finishing their review, the reviewer can:
+- Ask questions to better understand the change
+- Suggest changes to it
+- Approve the change
+
+Many changes go through a few rounds of reviews before being approved.
+
+### Approval and Merging
+
+During the review process, a successful change will accumulate "approvals" from
+satisfied reviewers. At some point, the change will have "enough approvals" and
+will be ready to merge. There's no hard-and-fast rule to determine when a
+change has enough approvals. Generally, simple and small changes are easy to
+approve, whereas changes that are breaking new ground, or have wide-ranging
+consequences, will require more reviews. If you're unsure as to whether a
+change is ready to merge, ask one of the maintainers.
+
+### Changes that never merge
+
+Some changes don't ever merge. A few good reasons why this may happen include:
+- Reviewers may find that the change isn't actually needed, is basically
+  unsound, or isn't aligned with the project's values or goals
+- Over the course of review, the contributor or reviewers may find a different
+  and better way to achieve the same goals
```

Comment:
I would add something along the lines of "In these cases, maintainers should openly communicate how their thinking or the situation has changed from the initial opening of the pull request, and make an effort to both thank the contributor, and look for opportunities to still involve him or her with future development."

The general idea is that you wouldn't want to say "OH sorry we don't need / want this anymore" but something more like "Our thinking has changed and we think this [other thing] might be a better strategy to pursue, would you be interested in helping with well scoped task]?

Reply:
This ☝️ is a really good idea. Concrete examples are really great IMO, especially to those unfamiliar with the process. 

A bit of a tangent, but: It can be really nerve-wracking to submit a pull request; one thing that is not at all nerve-wracking about submitting a PR to SourceCred is formatting your PR in terms of commit structure and message, because instructions and rationale are laid out clearly and comprehensively in the Contributing Guide and there's no grey area. We could aspire to something  similar in terms of etiquette; providing concrete examples about things to do in non-ideal circumstances feels like a way to remove grey area.

Reply:
It's probably implied, but pull request templates can help with a nice checklist of TODOs when opening the pull request, and specific groups to mention if the requester needs help with a particular issue. For example, I've used GitHub since 2011 but I still have cases when I'm not sure about a particular bullet point in a pull request template. The pull request template should have a catch all for this, because we just can't predict all situations / questions that might have a contributor hesitating and not sure about something. For example, if the requester is in this situation, he/she might be encouraged to open a draft pull request pinging some particular team that can give support before it's ready for review.

Reply:
Sounds good. Following on from [this comment above](https://github.com/sourcecred/docs/pull/1#discussion_r368335879), maybe we should have a "best practices for maintainers" section too. I'd rather see that in a followon PR so that this one can merge. :)

Reply:
Tracking suggestion here https://github.com/sourcecred/docs/issues/8